### PR TITLE
CS-1259 Match apprentice to apprenticeship using Accounts API

### PR DESCRIPTION
**Pull request by benagain**

### CODE REVIEW

File: src/SFA.DAS.ApprenticeCommitments.Api/Controllers/ApprenticeController.cs [hunk 1]
```diff
@@ -1,18 +1,26 @@
...
 using Microsoft.AspNetCore.Mvc;
-using SFA.DAS.ApprenticeCommitments.Apis.InnerApi;
+using SFA.DAS.ApprenticeCommitments.Apis.ApprenticeAccountsApi;
 using SFA.DAS.ApprenticeCommitments.Application.Services;
 using System;
 using System.Threading.Tasks;
 
 namespace SFA.DAS.ApprenticeCommitments.Api.Controllers
 {
+    public class TemporaryAccountsResponseReturningApiClient : ResponseReturningApiClient
```
Comment (Pauljgraham) [inline src/SFA.DAS.ApprenticeCommitments.Api/Controllers/ApprenticeController.cs:10]: I see what you mean, and yes the whole apprentice controller needs to be removed

**Review by Pauljgraham — COMMENTED** [review]
Let a few small comments. The only question is how are we creating the Apprentice record in AA because we can't do it here. Is it worth discussing this

File: src/SFA.DAS.ApprenticeCommitments/Apis/InnerApi/CreateApprenticeshipRequest.cs [hunk 2]
```diff
@@ -1,32 +1,23 @@
...
+﻿using SFA.DAS.SharedOuterApi.Interfaces;
+using System;
 
 namespace SFA.DAS.ApprenticeCommitments.Apis.InnerApi
 {
-    public class CreateApprenticeshipRequest : IPostApiRequest<CreateApprenticeshipRequestData>
+    internal class CreateApprenticeshipRequest : IPostApiRequest
     {
-        public string PostUrl => "/registrations";
+        public string PostUrl { get; }
```
[the remark this answers is not in the document]
Comment (benagain) [reply src/SFA.DAS.ApprenticeCommitments/Apis/InnerApi/CreateApprenticeshipRequest.cs:8]: Yes!

File: src/SFA.DAS.ApprenticeCommitments/Apis/InnerApi/CreateRegistrationRequest.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using SFA.DAS.SharedOuterApi.Interfaces;
+
+namespace SFA.DAS.ApprenticeCommitments.Apis.InnerApi
+{
+    public class CreateRegistrationRequest : IPostApiRequest<CreateRegistrationRequestData>
+    {
+        public string PostUrl => "/registrations";
```
[the remark this answers is not in the document]
Comment (benagain) [reply src/SFA.DAS.ApprenticeCommitments/Apis/InnerApi/CreateRegistrationRequest.cs:8]: That was modified in this file as part of #539 - I think there may be some merging ahead.

File: src/SFA.DAS.ApprenticeCommitments/Application/Services/ApprenticeCommitmentsService.cs [hunk 4]
```diff
@@ -14,9 +14,9 @@ public ApprenticeCommitmentsService(IInternalApiClient<ApprenticeCommitmentsConf
 
         public Task<bool> IsHealthy() => HealthCheck.IsHealthy(_client);
```
[the remark this answers is not in the document]
Comment (benagain) [reply src/SFA.DAS.ApprenticeCommitments/Application/Services/ApprenticeCommitmentsService.cs:15]: I don't think that the Apprentice Commitments outer API is responsible for the Apprentice Accounts inner API health.  It doesn't currently worry about all the other inner APIs it consumes.

**Review by Pauljgraham — APPROVED** [review]

**Pull request merged**